packages/itwinui-react/src/core/Tooltip/Tooltip.tsx

```diff
@@ -16,46 +63,116 @@ export type TooltipProps = {
+  //  */
+  // reference?: React.RefObject<Element> | null;
+  /**
+   * Element to portal tooltip to.
+   * Portals to ThemeProvider portalContainerRef by default.
+   */
+  portal?: boolean | { to: HTMLElement };
+};
 
-/**
- * Basic tooltip component to display informative content when an element is hovered or focused.
- * Uses the {@link Popover} component, which is a wrapper around [tippy.js](https://atomiks.github.io/tippyjs).
- * @example
- * <Tooltip content='tooltip text' placement='top'><div>Hover here</div></Tooltip>
- * @example
- * const buttonRef = React.useRef();
- * ...
- * <Button ref={buttonRef} />
- * <Tooltip content='tooltip text' reference={buttonRef} />
- */
-export const Tooltip = (props: TooltipProps) => {
-  const { content, children, className, style, visible, ref, id, ...rest } =
-    props;
+const useTooltip = ({
+  placement = 'top',
+  visible: controlledOpen,
+  toggleVisible: setControlledOpen,
+  followTrigger = false,
+}: TooltipOptions = {}) => {
+  const [uncontrolledOpen, setUncontrolledOpen] = React.useState(false);
 
-  return (
-    <Popover
-      visible={visible}
-      interactive={false}
-      content={
-        <Box
-          className={cx('iui-tooltip', className)}
-          style={style}
-          role='tooltip'
-          id={id}
-        >
-          {content}
-        </Box>
-      }
-      offset={[0, 4]}
-      ref={ref}
-      {...rest}
+  const open = controlledOpen ?? uncontrolledOpen;
+  const setOpen = setControlledOpen ?? setUncontrolledOpen;
+
+  const data = useFloating({
+    placement,
+    open,
+    onOpenChange: setOpen,
+    whileElementsMounted: (referenceEl, floatingEl, update) =>
+      autoUpdate(referenceEl, floatingEl, update, {
+        animationFrame: followTrigger,
+      }),
+    middleware: [offset(4), flip(), shift()],
+  });
+
+  const context = data.context;
+
+  const hover = useHover(context);
+  const focus = useFocus(context, {
+    enabled: controlledOpen == null,
+  });
+  const click = useClick(context);
+
+  const dismiss = useDismiss(context);
+  const role = useRole(context, { role: 'tooltip' });
+
+  const interactions = useInteractions([hover, focus, click, dismiss, role]);
+
+  return React.useMemo(
+    () => ({
+      open,
+      setOpen,
+      ...interactions,
+      ...data,
+    }),
+    [open, setOpen, interactions, data],
+  );
+};
+
+const TooltipComponent = ({
+  content,
+  children,
+  portal = false,
+  ...options
+}: TooltipOwnProps & TooltipOptions) => {
+  const tooltip = useTooltip(options);
+  const context = useGlobals();
+
+  const portalTo =
+    typeof portal !== 'boolean'
+      ? portal.to
+      : portal
+      ? context?.portalContainerRef?.current ?? getDocument()?.body
+      : null;
+
+  const contentBox = (
+    <Box
+      className={cx('iui-tooltip')}
+      ref={tooltip.refs.setFloating}
+      style={tooltip.floatingStyles}
+      {...tooltip.getFloatingProps()}
```

Comment:
missing forwardRef, cx, rest props, and need to merge style prop too

Reply:
Added